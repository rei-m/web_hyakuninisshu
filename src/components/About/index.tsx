import * as React from 'react';

const About = () => {
  return <div>サイトについて</div>;
};

export default About;
