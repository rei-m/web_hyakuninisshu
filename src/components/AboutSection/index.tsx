import * as React from 'react';
import styled from 'styled-components';
import { withTheme } from '../../styles';

const Root = withTheme(styled.div)`
  padding: ${({ theme }) => theme.spacing2x};
  box-sizing: border-box;
  max-width: 960px;
  margin: auto;
  text-align: left;
`;

const Section = withTheme(styled.section)`
  margin: ${({ theme }) => theme.spacing4x} 0;
`;

const SectionTitle = styled.h1`
  font-size: 2.4rem;
  position: relative;
  &:after {
    content: '';
    width: 100%;
    border-bottom: 1px solid #a9a9a9;
    position: absolute;
    bottom: -8px;
    left: 0;
  }
`;

const SectionText = withTheme(styled.div)`
  padding: ${({ theme }) => theme.spacing1x};
  line-height: 2rem;
`;

const AppBannerBox = styled.div`
  text-align: center;
`;

const AboutSection = () => (
  <Root>
    <Section>
      <SectionTitle>サイトについて</SectionTitle>
      <SectionText>
        このサイトは百人一首を覚えるためのサイトです。4択のクイズ形式で繰り返し練習することで効率よく百人一首を覚えることが出来ます。
      </SectionText>
    </Section>
    <Section>
      <SectionTitle>使い方について</SectionTitle>
      <SectionText>
        練習では様々な条件を組み合わせて出題範囲を指定できます。ご自身の習熟度に合わせて効率よく練習できます。
        もう全部覚えたという方は腕試しに挑戦しましょう。百首通しでランダムに出題されます。
        資料では百人一首の詳細な情報を閲覧できます。決まり字や歌の現代語訳などを確認することができます。
      </SectionText>
    </Section>
    <Section>
      <SectionTitle>アプリ版について</SectionTitle>
      <SectionText>
        Androidのみリリースしています。こちらは一旦インストールしていただけばオフラインでも使えます。iOS版は未定ですが、要望があればがんばります。
      </SectionText>
      <AppBannerBox>
        <a href="https://play.google.com/store/apps/details?id=me.rei_m.hyakuninisshu&hl=ja&pcampaignid=MKT-Other-global-all-co-prtnr-py-PartBadge-Mar2515-1">
          <img
            alt="Google Play で手に入れよう"
            src="https://play.google.com/intl/en_us/badges/images/generic/ja_badge_web_generic.png"
            style={{
              margin: 'auto',
              width: 200
            }}
          />
        </a>
      </AppBannerBox>
    </Section>
    <Section>
      <SectionTitle>運営者について</SectionTitle>
      <SectionText>
        当サイトは
        <a href="https://twitter.com/rei_m" target="_blank">
          @rei_m
        </a>
        が個人で運営しています。不具合や要望等あればTwitterまでご連絡ください。
        アプリのレビューで読み上げ機能が欲しいというお声をいただくのですが、使用可能ないい音源があれば搭載します。
      </SectionText>
    </Section>
  </Root>
);

export default AboutSection;
