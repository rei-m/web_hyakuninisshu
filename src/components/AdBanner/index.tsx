import * as React from 'react';
import AdSense from 'react-adsense';
import styled from '@src/styles/styled-components';

const Container = styled.div`
  padding: ${({ theme }) => theme.spacing2x};
`;

const AdBanner: React.FC<{}> = () => (
  <Container>
    <AdSense.Google
      client="ca-pub-4104372369598017"
      slot="9171068817"
      style={{ display: 'block' }}
      format="auto"
      responsive="true"
    />
  </Container>
);

export default AdBanner;
