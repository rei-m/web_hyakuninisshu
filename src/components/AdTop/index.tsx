import * as React from 'react';
import AdSense from 'react-adsense';
import styled from '@src/styles/styled-components';

const Container = styled.div`
  text-align: center;
  padding: ${({ theme }) => theme.spacing1x} 0;
`;

const Dummy = styled.div`
  width: 320px;
  height: 100px;
  border: 1px solid #123;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0 auto;
  @media (min-width: 500px) {
    width: 468px;
    height: 60px;
  }
  @media (min-width: 800px) {
    width: 728px;
    height: 90px;
  }
`;

const AdTop: React.FC<{ style?: React.CSSProperties }> = ({ style }) =>
  process.env.NODE_ENV === 'production' ? (
    <Container style={style}>
      <style
        dangerouslySetInnerHTML={{
          __html: `
  .ad_top { width: 320px; height: 100px; }
  @media(min-width: 500px) { .ad_top { width: 468px; height: 60px; } }
  @media(min-width: 800px) { .ad_top { width: 728px; height: 90px; } }　        
`,
        }}
      />
      <AdSense.Google
        className="ad_top"
        client="ca-pub-4104372369598017"
        slot="9171068817"
        style={{ display: 'inline-block' }}
        format=""
        responsive=""
      />
    </Container>
  ) : (
    <Container style={style}>
      <Dummy>この欄は広告枠です</Dummy>
    </Container>
  );

export default AdTop;
