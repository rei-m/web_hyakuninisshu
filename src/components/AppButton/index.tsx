import * as React from 'react';
import withStyles from '@material-ui/core/styles/withStyles';
import Icon from '@material-ui/core/Icon';
import Button from '@material-ui/core/Button';
import { appTheme } from '@src/styles/theme';

export interface Props {
  label: string;
  type: 'primary' | 'normal';
  icon?: string;
  style?: React.CSSProperties;
  onClick: () => void;
}

const styles = {
  button: {
    padding: '4px 15px',
    fontSize: '1.6rem',
    height: '40px',
  },
  normalButton: {
    color: '#182026',
    backgroundColor: '#f5f8fa',
    '&:active': {
      backgroundColor: appTheme.colorAccentActive,
    },
    '&:hover': {
      backgroundColor: '#ebf1f5',
    },
  },
  primaryButton: {
    color: '#ffffff',
    backgroundColor: appTheme.colorAccent,
    '&:active': {
      backgroundColor: appTheme.colorAccentActive,
    },
    '&:hover': {
      backgroundColor: appTheme.colorAccentHover,
    },
  },
  normalIcon: {
    marginRight: appTheme.spacing1x,
    color: '#5c7080',
  },
  primaryIcon: {
    marginRight: appTheme.spacing1x,
  },
};

const stylesProvider = () => styles;

type RenderProps = Props & {
  classes: {
    button: string;
    normalButton: string;
    primaryButton: string;
    normalIcon: string;
    primaryIcon: string;
  };
};

const AppButton = withStyles(stylesProvider)(({ label, type, icon, style, onClick, classes }: RenderProps) => {
  let buttonClassName = '';
  let iconClassName = '';
  if (type === 'primary') {
    buttonClassName = `${classes.button} ${classes.primaryButton}`;
    iconClassName = classes.primaryIcon;
  } else {
    buttonClassName = `${classes.button} ${classes.normalButton}`;
    iconClassName = classes.normalIcon;
  }

  return (
    <Button variant="outlined" onClick={onClick} className={buttonClassName} style={style}>
      {icon && (
        <Icon className={iconClassName} fontSize="small">
          {icon}
        </Icon>
      )}
      {label}
    </Button>
  );
});

export default AppButton;
