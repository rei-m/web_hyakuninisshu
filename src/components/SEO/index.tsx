import * as React from 'react';
import Helmet from 'react-helmet';
import { graphql, StaticQuery } from 'gatsby';
import { SiteMetaData } from '@src/types';

export interface Props {
  description?: string;
  lang?: string;
  meta?: Array<{
    name: string;
    content: string;
  }>;
  keywords?: string[];
  title: string;
}

export interface DefaultSEOQueryData {
  site: {
    siteMetadata: SiteMetaData;
  };
  ogpImage: {
    publicURL: string;
  };
}

const SEO: React.FC<Props> = ({ description, lang, meta, keywords, title }) => {
  const checkedLang: string = lang ? lang : 'ja';
  const checkedKeywords: string[] = keywords ? keywords : [];
  const checkedMeta: Array<{
    name: string;
    content: string;
  }> = meta ? meta : [];

  return (
    <StaticQuery
      query={query}
      render={(data: DefaultSEOQueryData) => {
        const checkedTitle = title || data.site.siteMetadata.title;
        const checkedDescription = description || data.site.siteMetadata.description;
        return (
          <Helmet
            htmlAttributes={{
              lang: checkedLang,
            }}
            title={checkedTitle}
            meta={[
              {
                name: `description`,
                content: checkedDescription,
              },
              {
                property: `og:title`,
                content: checkedTitle,
              },
              {
                property: `og:description`,
                content: checkedDescription,
              },
              {
                property: `og:type`,
                content: `website`,
              },
              {
                property: `og:image`,
                content: `https://hyakuninanki.net${data.ogpImage.publicURL}`,
              },
              {
                name: `twitter:card`,
                content: `summary`,
              },
              {
                name: `twitter:creator`,
                content: data.site.siteMetadata.author,
              },
              {
                name: `twitter:title`,
                content: checkedTitle,
              },
              {
                name: `twitter:description`,
                content: checkedDescription,
              },
              {
                name: `twitter:image`,
                content: `https://hyakuninanki.net${data.ogpImage.publicURL}`,
              },
            ]
              .concat(
                checkedKeywords.length > 0
                  ? {
                      name: `keywords`,
                      content: checkedKeywords.join(`, `),
                    }
                  : []
              )
              .concat(checkedMeta)}
          />
        );
      }}
    />
  );
};

export default SEO;

const query = graphql`
  query DefaultSEOQuery {
    site {
      siteMetadata {
        title
        description
        author
      }
    }
    ogpImage: file(relativePath: { eq: "app-icon.png" }) {
      publicURL
    }
  }
`;
