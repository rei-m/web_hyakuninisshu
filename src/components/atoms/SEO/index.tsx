import React from 'react';
import Helmet from 'react-helmet';
import { useSEO } from '@src/hooks/staticQueries/useSEO';

export type Props = {
  description?: string;
  lang?: string;
  meta?: Array<{
    name: string;
    content: string;
  }>;
  keywords?: string[];
  title: string;
};

const SEO = ({ description, lang = 'ja', meta = [], keywords = [], title }: Props) => {
  const { site, ogpImage } = useSEO();
  const checkedTitle = title || site.siteMetadata.title;
  const checkedDescription = description || site.siteMetadata.description;
  return (
    <Helmet
      htmlAttributes={{
        lang,
      }}
      title={checkedTitle}
      meta={[
        {
          name: `description`,
          content: checkedDescription,
        },
        {
          property: `og:title`,
          content: checkedTitle,
        },
        {
          property: `og:description`,
          content: checkedDescription,
        },
        {
          property: `og:type`,
          content: `website`,
        },
        {
          property: `og:image`,
          content: `https://hyakuninanki.net${ogpImage.publicURL}`,
        },
        {
          name: `twitter:card`,
          content: `summary`,
        },
        {
          name: `twitter:creator`,
          content: site.siteMetadata.author,
        },
        {
          name: `twitter:title`,
          content: checkedTitle,
        },
        {
          name: `twitter:description`,
          content: checkedDescription,
        },
        {
          name: `twitter:image`,
          content: `https://hyakuninanki.net${ogpImage.publicURL}`,
        },
      ]
        .concat(
          keywords.length > 0
            ? {
                name: `keywords`,
                content: keywords.join(`, `),
              }
            : []
        )
        .concat(meta)}
    />
  );
};

export default SEO;
