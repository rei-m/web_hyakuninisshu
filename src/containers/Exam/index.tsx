import * as React from 'react';
import { GlobalState } from '../../reducers/index';
import { branch, compose, renderComponent } from 'recompose';
import { withRouter, RouteComponentProps } from 'react-router-dom';
import { Answer, Question, ToriFuda } from '../../types';
import { connect, Dispatch } from 'react-redux';
import {
  answerQuestion,
  confirmCorrect,
  finishQuestions,
  openNextQuestion,
  restartQuestions
} from '../../actions/questions';
import QuestionSection, {
  QuestionSectionProps
} from '../../components/QuestionSection';
import QuestionCorrect, {
  QuestionCorrectProps
} from '../../components/QuestionCorrect';
import ExamResult, { ExamResultProps } from '../../components/ExamResult';
import ExamInitializer from '../ExamInitializer';
import { QuestionState } from '../../enums';

export interface ExamOwnProps {
  readonly started: boolean;
  readonly questions: Question[];
  readonly answers: Answer[];
  readonly questionState?: QuestionState;
}

export type ExamConnectedProps = Omit<
  QuestionSectionProps,
  'onClickToriFuda' | 'onClickResult'
>;

export type ExamDispatchProps = Pick<ExamResultProps, 'onClickRestart'> &
  Pick<QuestionCorrectProps, 'onClickGoToNext' | 'onClickGoToResult'> &
  Pick<QuestionSectionProps, 'onClickToriFuda' | 'onClickResult'>;

export type ExamProps = ExamOwnProps & ExamConnectedProps & ExamDispatchProps;

const mapStateToProps = (
  { questionsState }: GlobalState,
  { location }: RouteComponentProps<{}>
): ExamOwnProps & ExamConnectedProps => {
  const { submitTime } = location.state;
  const {
    answers,
    currentIndex,
    lastStartedTime,
    questions,
    questionState
  } = questionsState;

  return {
    answer: answers[currentIndex],
    answers,
    currentPosition: currentIndex + 1,
    question: questions[currentIndex],
    questionState,
    questions,
    started: !!lastStartedTime && lastStartedTime > submitTime,
    totalCount: questions.length
  };
};

const mapDispatchToProps = (
  dispatch: Dispatch<GlobalState>
): ExamDispatchProps => {
  return {
    onClickGoToNext: () => {
      dispatch(openNextQuestion());
    },
    onClickGoToResult: () => {
      dispatch(finishQuestions());
    },
    onClickRestart: () => {
      dispatch(restartQuestions());
    },
    onClickResult: () => {
      dispatch(confirmCorrect());
    },
    onClickToriFuda: ({ questionId, karutaId }: ToriFuda) => {
      dispatch(answerQuestion(questionId, karutaId));
    }
  };
};

const isStarted = ({ started }: ExamOwnProps) => started;

const withStartedCheck = branch<ExamOwnProps>(
  isStarted,
  component => component,
  renderComponent(ExamInitializer)
);

const isConfirmedQuestionResult = ({ questionState }: ExamOwnProps) =>
  questionState === QuestionState.ConfirmCorrect;

const renderQuestionCorrect = ({
  question,
  questions,
  answers,
  onClickGoToNext,
  onClickGoToResult
}: ExamProps) => {
  const { correctKaruta } = question;
  return (
    <QuestionCorrect
      karuta={correctKaruta}
      isAllAnswered={questions.length === answers.length}
      onClickGoToNext={onClickGoToNext}
      onClickGoToResult={onClickGoToResult}
    />
  );
};

const withConfirmedQuestionResultCheck = branch<ExamOwnProps>(
  isConfirmedQuestionResult,
  renderComponent(renderQuestionCorrect),
  component => component
);

const isFinished = ({ questionState }: ExamOwnProps) =>
  questionState === QuestionState.Finished;

const renderResult = ({
  questions,
  answers,
  onClickRestart,
  totalCount
}: ExamProps) => {
  const correctCount = answers.filter(a => a.correct).length;
  const averageAnswerSecond =
    answers.reduce((prev, current) => prev + current.time, 0) /
    1000 /
    totalCount;
  return (
    <ExamResult
      averageAnswerSecond={Math.round(averageAnswerSecond * 100) / 100}
      totalCount={totalCount}
      correctCount={correctCount}
      answers={answers}
      questions={questions}
      onClickRestart={onClickRestart}
    />
  );
};

const withFinishedCheck = branch<ExamOwnProps>(
  isFinished,
  renderComponent(renderResult),
  component => component
);

const ExamIndex = compose<ExamProps, ExamProps>(
  withStartedCheck,
  withConfirmedQuestionResultCheck,
  withFinishedCheck
)(QuestionSection);

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(ExamIndex)
);
