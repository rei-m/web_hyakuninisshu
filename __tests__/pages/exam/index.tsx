import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import Exam from '@src/pages/exam';
import { DefaultSEOQueryData } from '@src/components/SEO';
import { setUpQueryOnce } from '@test/helpers';

describe('/exam', () => {
  beforeEach(() => {
    setUpQueryOnce<DefaultSEOQueryData>({
      site: {
        siteMetadata: {
          title: 'exam title',
          description: 'exam description',
          author: '@rei-m',
        },
      },
      ogpImage: {
        publicURL: 'http://localhost:8000/app.png',
      },
    });
  });

  it('should render component', () => {
    const renderer = ReactTestRenderer.create(<Exam />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
