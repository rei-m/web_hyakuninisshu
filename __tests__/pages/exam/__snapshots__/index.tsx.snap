// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/exam should render component 1`] = `
Array [
  <header
    className="sc-bwzfXH geYBAS"
  >
    <div
      className="sc-bdVaJa flBAph"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <span
        className="sc-bxivhb iUaUQC"
        color="#fff"
        data-test="back"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-1"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </span>
    </div>
    <h1
      className="sc-htpNat eSnwWs"
    >
      百人一首 - 腕試し -
    </h1>
  </header>,
  <div
    className="sc-gzVnrw hzJxWQ"
  >
    <section
      className="sc-fjdhpX gNpQzj"
    >
      <h1
        className="sc-iwsKbI jkmlvZ"
      >
        腕試し
      </h1>
      <div
        className="sc-gZMcBi iXZiBx"
      >
        <div
          className="sc-gqjmRU cCYYjG"
        >
          この欄は広告枠です
        </div>
      </div>
      <div
        className="sc-jzJRlG kwxYRR"
      >
        全百首からランダムに出題されます。
        <br />
        練習の成果を確認しましょう。
      </div>
      <button
        className="MuiButtonBase-root-41 MuiButton-root-15 MuiButton-outlined-23 Component-button-10 Component-primaryButton-12"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "marginBottom": "16px",
            "marginTop": "32px",
          }
        }
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-16"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-1 MuiSvgIcon-fontSizeSmall-8 Component-primaryIcon-14"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
          腕試しをはじめる
        </span>
        <span
          className="MuiTouchRipple-root-44"
        />
      </button>
      <div
        className="sc-VigVT ieiGAu"
      >
        <div
          className="sc-jTzLTM ljXwZC"
        >
          この欄は広告枠です
        </div>
      </div>
    </section>
  </div>,
  <nav
    className="sc-EHOje extRj"
  >
    <span
      className="sc-bZQynM jmnWoK"
    >
      <div
        className="sc-bdVaJa flBAph"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          color="#fff"
          href="/training"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <span
            className="sc-ifAKCX xOgRI"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-1"
              focusable="false"
              role="presentation"
              style={
                Object {
                  "fontSize": undefined,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </a>
      </div>
    </span>
    <span
      className="sc-bZQynM jmnWoK"
    >
      <div
        className="sc-bdVaJa flBAph"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          color="#fff"
          href="/exam"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <span
            className="sc-ifAKCX jHOBkV"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-1"
              focusable="false"
              role="presentation"
              style={
                Object {
                  "fontSize": undefined,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M22 10l-6-6H4c-1.1 0-2 .9-2 2v12.01c0 1.1.9 1.99 2 1.99l16-.01c1.1 0 2-.89 2-1.99v-8zm-7-4.5l5.5 5.5H15V5.5z"
              />
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
            </svg>
          </span>
        </a>
      </div>
    </span>
    <span
      className="sc-bZQynM jmnWoK"
    >
      <div
        className="sc-bdVaJa flBAph"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          color="#fff"
          href="/karutas"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <span
            className="sc-ifAKCX fGzQGR"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-1"
              focusable="false"
              role="presentation"
              style={
                Object {
                  "fontSize": undefined,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm16-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-1 9H9V9h10v2zm-4 4H9v-2h6v2zm4-8H9V5h10v2z"
              />
            </svg>
          </span>
        </a>
      </div>
    </span>
    <span
      className="sc-bZQynM jmnWoK"
    >
      <div
        className="sc-bdVaJa flBAph"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          color="#fff"
          href="/about"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <span
            className="sc-ifAKCX btkJVi"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-1"
              focusable="false"
              role="presentation"
              style={
                Object {
                  "fontSize": undefined,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h20v20H0V0z"
                fill="none"
                transform="scale(1.2, 1.2)"
              />
              <path
                d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
                transform="scale(1.2, 1.2)"
              />
            </svg>
          </span>
        </a>
      </div>
    </span>
  </nav>,
]
`;
